backend/inspirehep/hal/tasks.py

```diff
@@ -65,12 +68,30 @@ def _hal_push(record):
         ids = record.get("external_system_identifiers", [])
         hal_value = get_values_for_schema(ids, "HAL")
         hal_id = hal_value[0] if hal_value else ""
-
         receipt = None
-        if hal_id:
-            receipt = update(tei, hal_id)
-            LOGGER.info("HAL updated.", recid=record["control_number"], hal_id=hal_id)
-        else:
-            receipt = create(tei)
-            LOGGER.info("HAL created.", recid=record["control_number"], receipt=receipt)
+        try:
```

Comment:
these can be easily split into two functions `hal_create` and `had_update`, the logic would be easier to test and maintain without the `if/else` `try/except`  spaghetti.